include/benchmark/benchmark.h

```diff
@@ -429,16 +430,18 @@ class State {
   // Returns true if the benchmark should continue through another iteration.
   // NOTE: A benchmark may not return from the test until KeepRunning() has
   // returned false.
-  bool KeepRunning() {
-    if (BENCHMARK_BUILTIN_EXPECT(!started_, false)) {
-      StartKeepRunning();
-    }
-    bool const res = (--total_iterations_ != 0);
-    if (BENCHMARK_BUILTIN_EXPECT(!res, false)) {
-      FinishKeepRunning();
-    }
-    return res;
-  }
+  bool KeepRunning();
+
+  // Returns true iff the benchmark should run n more iterations.
+  // NOTE: A benchmark must not return from the test until KeepRunningBatch()
```

Comment:
We should note the requirement that `n != 0` since we'll enter an infinite loop if a user passes `0`. And users are silly, they might attempt to do that.

Reply:
I'll send a PR to assert and note it.

Reply:
Fixes are in https://github.com/google/benchmark/pull/526.